geode-core/src/main/java/org/apache/geode/internal/cache/BucketSetHelper.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache;
+
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+public class BucketSetHelper {
+  public static int get(int[] bucketSet, int index) {
+    return bucketSet[index + 1];
+  }
+
+  public static int length(int[] bucketSet) {
+    return null == bucketSet || bucketSet.length < 2 ? 0 : bucketSet[0];
+  }
+
+  public static void add(int[] bucketSet, int value) {
```

Comment:
What if the bucketSet doesn't have capacity? Should it resize and return a new array?

Reply:
Before add method is used, array is allocated, with enough capacity, so it should never happen that we don`t have capacity. If it happens, it is fault.